core/src/script.rs

```diff
@@ -13,54 +13,13 @@ use std::mem;
-    // the script to run in binary part; however, a common use case is
-    // that CKB would provide common system cells containing common verification
-    // algorithm, such as P2SH-SHA3-SECP256K1, in the meantime, crypto experts might
-    // also put alternative advanced verfication algorithms on the chain. So another
-    // way of loading a script, is that reference can be used to specify
-    // an existing cell, when CKB runs the script, CKB will load the script from the
-    // existing cell. This has the benefit of promoting code reuse, and reducing
-    // transaction size: a typical secp256k1 verfication algorithm can take 1.2 MB
-    // in space, which is not ideal to put in every tx input.
-    // Note that the referenced cell here might also be included in transaction's
-    // deps part, otherwise CKB will fail to verify the script.
-    // CKB only enforces that reference and binary cannot both be
-    // None, when they both contains actual value(though this is not recommended),
-    // binary will be used.
-    // When calculating script type hash, reference, binary,
-    // and signed_args will all be included.
-    pub reference: Option<H256>,
-    pub binary: Option<Vec<u8>>,
-    // Pre-defined arguments that are considered part of the script.
-    // When signed_args contains <a>, <b>, and args contains <c>,
-    // <d>, <e>, binary will then be executed with arguments <a>, <b>,
-    // <c>, <d>, <e>.
-    // This can be useful when binary is fixed, but depending on different
-    // use case, we might have different initial parameters. For example, in
-    // secp256k1 verification, we need to provide pubkey first, this cannot be
-    // part of arguments, otherwise users can provide signatures signed by
-    // arbitrary private keys. On the other hand, include pubkey inside
-    // binary is not good for distribution, since the script here can
-    // be over 1 megabytes. So signed_args helps here to preserve one common
-    // binary, while enable us to provide different pubkeys for different
-    // transactions.
-    // For most verification algorithms, args will contain the signature
-    // and any additional parameters needed by cell validator, while
-    // signed_args will contain pubkey used in the signing part.
-    pub signed_args: Vec<Vec<u8>>,
-}
-
-struct OptionDisplay<T>(Option<T>);
-
-impl<T: fmt::Display> fmt::Display for OptionDisplay<T> {
-    fn fmt(&self, f: &mut fmt::Formatter) -> fmt::Result {
-        match self.0 {
-            Some(ref v) => write!(f, "Some({})", v),
-            None => write!(f, "None"),
-        }
-    }
+    // The reference hash here can be used to refer to binary in any of the
+    // following locations:
+    // 1. Data part of a dep cell in current transaction, reference hash
+    // must be the hash of the cell data so as to reference this cell
+    // 2. An embed item in current transaction, reference hash must be the
```

Comment:
So the embed item is a one-time data?

Reply:
Yep, it's like the original binary but at a different place with more potential use cases